src/domain/hanke/edit/utils.test.ts

```diff
@@ -0,0 +1,42 @@
+import { HankeAlueFormState } from './types';
+import { getAreaDefaultName } from './utils';
+
+function getArea(areaName: string): HankeAlueFormState {
+  return {
+    nimi: areaName,
+    id: null,
+    haittaAlkuPvm: '2023-01-12T00:00:00Z',
+    haittaLoppuPvm: '2024-11-27T00:00:00Z',
+    kaistaHaitta: null,
+    kaistaPituusHaitta: null,
+    meluHaitta: null,
+    polyHaitta: null,
+    tarinaHaitta: null,
+  };
+}
+
+test('Should get default area name when there are no areas', () => {
+  const areaName = getAreaDefaultName([]);
+
+  expect(areaName).toBe('Hankealue 1');
+});
+
+test('Should get correct default area name based on other areas', () => {
```

Comment:
This could also use one test where the custom name has the largest number. E.g.

```
  const areaName = getAreaDefaultName([
    getArea('Hankealue 1'),
    getArea('Hankealue 3'),
    getArea('Hankealue 7'),
    getArea('Oma alue 9'),
  ]);


  expect(areaName).toBe('Hankealue 8');
```

Reply:
Good point. Added test.